cvmfs/platform_linux.h

```diff
@@ -318,22 +318,30 @@ inline std::string platform_getexepath() {
   return "";
 }
 
-inline uint64_t platform_monotonic_time() {
+inline struct timespec _time_with_clock(int clock) {
   struct timespec tp;
+  int retval = clock_gettime(clock, &tp);
+  assert(retval == 0);
+  return tp;
+}
+
+inline uint64_t platform_monotonic_time() {
 #ifdef CLOCK_MONOTONIC_COARSE
-  int retval = clock_gettime(CLOCK_MONOTONIC_COARSE, &tp);
+  struct timespec tp = _time_with_clock(CLOCK_MONOTONIC_COARSE);
 #else
-  int retval = clock_gettime(CLOCK_MONOTONIC, &tp);
+  struct timespec tp = _time_with_clock(CLOCK_MONOTONIC);
 #endif
-  assert(retval == 0);
   return tp.tv_sec + (tp.tv_nsec >= 500000000);
 }
 
 inline uint64_t platform_monotonic_time_ns() {
-  struct timespec tp;
-  int retval = clock_gettime(CLOCK_MONOTONIC, &tp);
-  assert(retval == 0);
-  return static_cast<uint64_t>(tp.tv_sec*1e9 + tp.tv_nsec);
+  struct timespec tp = _time_with_clock(CLOCK_MONOTONIC);
+  return static_cast<uint64_t>(tp.tv_sec * 1e9 + tp.tv_nsec);
+}
+
+inline uint64_t platform_time() {
```

Comment:
We should add the unit and the time source to the function name.  I'd suggest `platform_realtime_ns`